### PR TITLE
MAINT: update README.rst.  Fix broken links and add "what is PyWavelets"

### PR DESCRIPTION
**Pull request by rgommers**
Also reorganized everything a bit and remove a not very meaningful usage example.

[ci skip]

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -29,51 +24,42 @@ The main features of PyWavelets are:
...
+`python-wavelets`, `python-pywt` or a similar package name.
 
-Binary packages for several Linux distributors are maintained by Open Source
-community contributors. Query your Linux package manager tool
-for `python-pywavelets`, `python-wavelets`, `python-pywt` or similar package name.
+- Install PyWavelets with ``pip install PyWavelets``.
 
-.. seealso::  :ref:`Development notes <dev-index>` section contains more
-              information on building and installing from source code.
+- To build and install from source, navigate to downloaded PyWavelets source
```
Comment (grlee77) [inline README.rst]: "downloaded" -> "the downloaded" or just "the"

File: README.rst [hunk 2]
```diff
@@ -29,51 +24,42 @@ The main features of PyWavelets are:
...
+  code directory and type ``python setup.py install``.
 
-Documentation
--------------
+The most recent *development* version can be found on GitHub at
+https://github.com/PyWavelets/pywt.
 
-Documentation with detailed examples and links to more resources is available
-online at http://pywavelets.readthedocs.org.
+The latest release, including source and binary package for Windows, is
```
Comment (grlee77) [inline README.rst]: package -> packages

**Comment by grlee77** [comment]
just a couple nitpicks

**Comment by rgommers** [comment]
Thanks, fixed.

**Pull request merged**